internal/db/resource.go

```diff
@@ -5,67 +5,61 @@ import (
 	"context"
 	"database/sql"
 
+	"gorm.io/gorm"
+
+	"github.com/canonical/jimm/v3/internal/dbmodel"
 	"github.com/canonical/jimm/v3/internal/errors"
 	"github.com/canonical/jimm/v3/internal/servermon"
 )
 
-// RESOURCES_RAW_SQL contains the raw query fetching entities from multiple tables, with their respective entity parents.
-const RESOURCES_RAW_SQL = `
-(
-	SELECT 'application_offer' AS type, 
-			application_offers.uuid AS id, 
-			application_offers.name AS name, 
-			models.uuid AS parent_id,
-			models.name AS parent_name,
-			'model' AS parent_type
-	FROM application_offers
-	JOIN models ON application_offers.model_id = models.id
-)
-UNION
-(
-	SELECT 'cloud' AS type, 
-			clouds.name AS id, 
-			clouds.name AS name, 
-			'' AS parent_id,
-			'' AS parent_name,
-			'' AS parent_type
-	FROM clouds
-)
-UNION
-(
-	SELECT 'controller' AS type, 
-		controllers.uuid AS id, 
-		controllers.name AS name, 
-		'' AS parent_id,
-		'' AS parent_name,
-		'' AS parent_type
-		FROM controllers
-)
-UNION
-(
-	SELECT 'model' AS type, 
-			models.uuid AS id, 
-			models.name AS name, 
-			controllers.uuid AS parent_id,
-			controllers.name AS parent_name,
-			'controller' AS parent_type
-	FROM models
-	JOIN controllers ON models.controller_id = controllers.id
-)
-UNION
-(
-	SELECT 'service_account' AS type, 
-			identities.name AS id, 
-			identities.name AS name, 
-			'' AS parent_id,
-			'' AS parent_name,
-			'' AS parent_type
-	FROM identities
-	WHERE name LIKE '%@serviceaccount'
-)
-ORDER BY type, id
-OFFSET ?
-LIMIT  ?;
+const ApplicationOffersQueryKey = "application_offers"
+const SELECT_APPLICATION_OFFERS = `
```

Comment:
I'd rather name them like this:

```
const ApplicationOffersQueryKey
const ApplicationOffersQuery
```

What do you think?

Reply:
it's a key indeed, not a query. So i don't know about this change

Reply:
+1 Babak's suggestion, I don't think the all caps naming for constants is very helpful, I'd reserve it for env vars.

Reply:
i deleted my comment because i though it was referring to the key and not the query itself

Reply:
i deleted my comment, because i though babak was talking about changing the key name